File: app/src/main/java/com/zoo/dongyeah/ActivityMain.java
package com.zoo.dongyeah;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.content.pm.Signature;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.util.Base64;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import net.daum.mf.map.api.CalloutBalloonAdapter;
import net.daum.mf.map.api.CameraUpdateFactory;
import net.daum.mf.map.api.MapLayout;
import net.daum.mf.map.api.MapPOIItem;
import net.daum.mf.map.api.MapPoint;
import net.daum.mf.map.api.MapPointBounds;
import net.daum.mf.map.api.MapView;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class ActivityMain extends AppCompatActivity implements MapView.MapViewEventListener, MapView.POIItemEventListener {
    private static final MapPoint CUSTOM_MARKER_POINT = MapPoint.mapPointWithGeoCoord(37.537229, 127.005515);
    private static final MapPoint CUSTOM_MARKER_POINT2 = MapPoint.mapPointWithGeoCoord(37.447229, 127.015515);
    private static final MapPoint DEFAULT_MARKER_POINT = MapPoint.mapPointWithGeoCoord(37.4020737, 127.1086766);

    private MapView mMapView;
    private MapPOIItem mDefaultMarker;
    private static final String LOG_TAG = "com.zoo.dongyeah";

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mMapView = (MapView) findViewById(R.id.map_view);
        mMapView.setDaumMapApiKey(BuildConfig.KAKAO_API_KEY);
        mMapView.setMapViewEventListener(this);
        mMapView.setPOIItemEventListener(this);

        createDefaultMarker(mMapView);
        showAll();


        mMapView.setCurrentLocationTrackingMode(MapView.CurrentLocationTrackingMode.TrackingModeOnWithoutHeading);



    }

    @Override
    public void onMapViewInitialized(MapView mapView) {

    }


    private void createDefaultMarker(MapView mapView) {
        mDefaultMarker = new MapPOIItem();
        String name = "Default Marker";
        mDefaultMarker.setItemName(name);
        mDefaultMarker.setTag(0);
        mDefaultMarker.setMapPoint(DEFAULT_MARKER_POINT);
        mDefaultMarker.setMarkerType(MapPOIItem.MarkerType.BluePin);
        mDefaultMarker.setSelectedMarkerType(MapPOIItem.MarkerType.RedPin);

        mapView.addPOIItem(mDefaultMarker);
        mapView.selectPOIItem(mDefaultMarker, true);
        mapView.setMapCenterPoint(DEFAULT_MARKER_POINT, false);
    }

    private void showAll() {
        int padding = 20;
        float minZoomLevel = 7;
        float maxZoomLevel = 10;
        MapPointBounds bounds = new MapPointBounds(CUSTOM_MARKER_POINT, DEFAULT_MARKER_POINT);
        mMapView.moveCamera(CameraUpdateFactory.newMapPointBounds(bounds, padding, minZoomLevel, maxZoomLevel));
    }

    @Override
    public void onMapViewCenterPointMoved(MapView mapView, MapPoint mapPoint) {

    }

    @Override
    public void onMapViewZoomLevelChanged(MapView mapView, int i) {

    }

    @Override
    public void onMapViewSingleTapped(MapView mapView, MapPoint mapPoint) {

    }

    @Override
    public void onMapViewDoubleTapped(MapView mapView, MapPoint mapPoint) {

    }

    @Override
    public void onMapViewLongPressed(MapView mapView, MapPoint mapPoint) {

    }

    @Override
    public void onMapViewDragStarted(MapView mapView, MapPoint mapPoint) {

    }

    @Override
    public void onMapViewDragEnded(MapView mapView, MapPoint mapPoint) {

    }

    @Override
    public void onMapViewMoveFinished(MapView mapView, MapPoint mapPoint) {

    }

    @Override
    public void onPOIItemSelected(MapView mapView, MapPOIItem mapPOIItem) {

    }

    @Override
    public void onCalloutBalloonOfPOIItemTouched(MapView mapView, MapPOIItem mapPOIItem) {
        Toast.makeText(this, "Clicked " + mapPOIItem.getItemName() + " Callout Balloon", Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onCalloutBalloonOfPOIItemTouched(MapView mapView, MapPOIItem mapPOIItem, MapPOIItem.CalloutBalloonButtonType calloutBalloonButtonType) {

    }

    @Override
    public void onDraggablePOIItemMoved(MapView mapView, MapPOIItem mapPOIItem, MapPoint mapPoint) {

    }
}

File: app/src/main/java/com/zoo/dongyeah/ActivityMain.kt
package com.zoo.dongyeah

import androidx.appcompat.app.AppCompatActivity
import net.daum.mf.map.api.MapView.POIItemEventListener
import android.os.Bundle
import android.util.Log
import android.view.View
import com.zoo.dongyeah.R
import com.zoo.dongyeah.ActivityMain
import android.widget.Toast
import net.daum.mf.map.api.*
import net.daum.mf.map.api.MapPOIItem.CalloutBalloonButtonType
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import kotlin.math.log

class ActivityMain : AppCompatActivity(), MapView.MapViewEventListener, POIItemEventListener {
    private var mMapView: MapView? = null
    private var mDefaultMarker: MapPOIItem? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        getHospitalData()

        mMapView = findViewById<View>(R.id.map_view) as MapView
        mMapView!!.setDaumMapApiKey(BuildConfig.KAKAO_API_KEY)
        mMapView!!.setMapViewEventListener(this)
        mMapView!!.setPOIItemEventListener(this)
        createDefaultMarker(mMapView)
        showAll()
        mMapView!!.currentLocationTrackingMode =
            MapView.CurrentLocationTrackingMode.TrackingModeOnWithoutHeading
    }

    override fun onMapViewInitialized(mapView: MapView) {}
    private fun createDefaultMarker(mapView: MapView?) {
        mDefaultMarker = MapPOIItem()
        val name = "Default Marker"
        mDefaultMarker!!.itemName = name
        mDefaultMarker!!.tag = 0
        mDefaultMarker!!.mapPoint = DEFAULT_MARKER_POINT
        mDefaultMarker!!.markerType = MapPOIItem.MarkerType.BluePin
        mDefaultMarker!!.selectedMarkerType = MapPOIItem.MarkerType.RedPin
        mapView!!.addPOIItem(mDefaultMarker)
        mapView.selectPOIItem(mDefaultMarker, true)
        mapView.setMapCenterPoint(DEFAULT_MARKER_POINT, false)
    }

    private fun showAll() {
        val padding = 20
        val minZoomLevel = 7f
        val maxZoomLevel = 10f
        val bounds = MapPointBounds(CUSTOM_MARKER_POINT, DEFAULT_MARKER_POINT)
        mMapView!!.moveCamera(
            CameraUpdateFactory.newMapPointBounds(
                bounds,
                padding,
                minZoomLevel,
                maxZoomLevel
            )
        )
    }

    override fun onMapViewCenterPointMoved(mapView: MapView, mapPoint: MapPoint) {}
    override fun onMapViewZoomLevelChanged(mapView: MapView, i: Int) {}
    override fun onMapViewSingleTapped(mapView: MapView, mapPoint: MapPoint) {}
    override fun onMapViewDoubleTapped(mapView: MapView, mapPoint: MapPoint) {}
    override fun onMapViewLongPressed(mapView: MapView, mapPoint: MapPoint) {}
    override fun onMapViewDragStarted(mapView: MapView, mapPoint: MapPoint) {}
    override fun onMapViewDragEnded(mapView: MapView, mapPoint: MapPoint) {}
    override fun onMapViewMoveFinished(mapView: MapView, mapPoint: MapPoint) {}
    override fun onPOIItemSelected(mapView: MapView, mapPOIItem: MapPOIItem) {}
    override fun onCalloutBalloonOfPOIItemTouched(mapView: MapView, mapPOIItem: MapPOIItem) {
        Toast.makeText(
            this,
            "Clicked " + mapPOIItem.itemName + " Callout Balloon",
            Toast.LENGTH_SHORT
        ).show()
    }

    override fun onCalloutBalloonOfPOIItemTouched(
        mapView: MapView,
        mapPOIItem: MapPOIItem,
        calloutBalloonButtonType: CalloutBalloonButtonType
    ) {
    }

    override fun onDraggablePOIItemMoved(
        mapView: MapView,
        mapPOIItem: MapPOIItem,
        mapPoint: MapPoint
    ) {
    }

    private fun getHospitalData() {
        val service = RetrofitClient.getInstance().create(HospitalAPI::class.java)
        val call = service.getData(127.005515, 37.537229, 3000)
        //val call = service.getData()
        call.enqueue(object : Callback<HospitalData> {
            override fun onResponse(call: Call<HospitalData>, response: Response<HospitalData>) {
                if (response.isSuccessful) {
                }
            }

            override fun onFailure(call: Call<HospitalData>, t: Throwable) {
                Log.e(LOG_TAG, t.printStackTrace().toString())
                Log.e(LOG_TAG, t.message)
            }

        })
    }

    companion object {
        private val CUSTOM_MARKER_POINT = MapPoint.mapPointWithGeoCoord(37.537229, 127.005515)
        private val CUSTOM_MARKER_POINT2 = MapPoint.mapPointWithGeoCoord(37.447229, 127.015515)
        private val DEFAULT_MARKER_POINT = MapPoint.mapPointWithGeoCoord(37.4020737, 127.1086766)
        private const val LOG_TAG = "com.zoo.dongyeah"
    }
}